### PR TITLE
initial how-to for TLS service deployment ( and small addition to daemon ssl urls )

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: docs/synapse/devops.rst [hunk 1]
```diff
@@ -6,7 +6,63 @@
...
+-------------------
+
+For production deployements, it is recommended that all services use the built in ``telepath`` SSL/TLS
+protocol.  You may deploy a service using TLS encryption by specifying a ``--telepath`` listen URL option, such
+as ``ssl://cortex.vertex.link/``.  If it is necessary to separate the socket bind address from the hostname,
+a hostname query parameter may be specified: ``ssl://0.0.0.0/?hostname=cortex.vertex.link``
+
+The client will also need to specify an SSL/TLS ``telepath`` URL such as: ``ssl://visi:passwd@cortex.vertex.link``.
+
+Once the ``ssl`` protocol is specified, the ``hostname``, either from a query parameter or from the URL's
```
Comment (jnwatson) [inline docs/synapse/devops.rst:22]: move 'the hostname' after 'network location,'

File: docs/synapse/devops.rst [hunk 2]
```diff
@@ -6,7 +6,63 @@
...
+network location, is used to lookup a matching ``crt`` and ``key`` file pair from your server certificate directory
+located at ``~/.syn/certs/hosts``.  See the following sections for how to setup server certificates.
+
+Self-Signed Certificates using certtool
+***************************************
+
+For self-signed certificates, we need to generate a CA certificate and key as well as a server certificate and key.
+
+The synapse ``certtool`` can be used to easily generate a CA and server certificates. For example, we we wanted
```
Comment (jnwatson) [inline docs/synapse/devops.rst:31]: "we we". -> "If we"?

File: docs/synapse/devops.rst [hunk 3]
```diff
@@ -6,7 +6,63 @@
...
 ####################
 
-TBD - See :ref:`quickstart` for current docs.
+General
+=======
+
+TLS/SSL Deployments
+-------------------
+
+For production deployements, it is recommended that all services use the built in ``telepath`` SSL/TLS
```
Comment (jnwatson) [inline docs/synapse/devops.rst:15]: "deployements"

Comment (jnwatson) [inline docs/synapse/devops.rst:15]: "built in" -> "built-in"

File: docs/synapse/devops.rst (hunk 1 above)
Comment (jnwatson) [inline docs/synapse/devops.rst:22]: URL's -> URLs

File: docs/synapse/devops.rst [hunk 4]
```diff
@@ -6,7 +6,63 @@
...
+protocol.  You may deploy a service using TLS encryption by specifying a ``--telepath`` listen URL option, such
+as ``ssl://cortex.vertex.link/``.  If it is necessary to separate the socket bind address from the hostname,
```
Comment (jnwatson) [inline docs/synapse/devops.rst:17]: when would this be necessary? (And put that text in)

File: docs/synapse/devops.rst [hunk 5]
```diff
@@ -6,7 +6,63 @@
...
+    key saved: /home/visi/.syn/certs/cas/vertex.key
+    cert saved: /home/visi/.syn/certs/cas/vertex.crt
+
+We can then generate a server certificate and keyfile pair and sign it with our new CA::
+    python -m synapse.tools.easycert --server cortex.vertex.link --signas vertex
+    key saved: /home/visi/.syn/certs/hosts/cortex.vertex.link.key
+    cert saved: /home/visi/.syn/certs/hosts/cortex.vertex.link.crt
+
+To verify the server certificate, clients will need to have the ``~/.syn/certs/cas/vertex.crt`` file in their
+certificate directory (NOTE: do not distribute the ``~/.syn/certs/cas/vertex.key`` file as that would allow regular
```
Comment (jnwatson) [inline docs/synapse/devops.rst:43]: No need to put in paretheses.

File: docs/synapse/devops.rst (hunk 3 above)
Comment (invisig0th) [reply docs/synapse/devops.rst:15]: lol, hadn't turned on ``:set spell`` yet :D

File: docs/synapse/devops.rst (hunk 4 above)
Comment (invisig0th) [reply docs/synapse/devops.rst:17]: There are a zillion potential reasons for this, not the least of which is him not actually having a public IP address actually assigned to his interface.  I'll make a bit more, but on the whole, the "my bind address does not match my hostname" discussion is out of scope.

File: docs/synapse/devops.rst (hunk 1 above)
Comment (invisig0th) [reply docs/synapse/devops.rst:22]: wouldn't be grammatically correct.  lets discuss.

Comment (invisig0th) [reply docs/synapse/devops.rst:22]: it is the network location that belongs to the URL not a plural URL...

File: docs/synapse/devops.rst [hunk 6]
```diff
@@ -6,7 +6,69 @@
...
+=======
+
+TLS/SSL Deployments
+-------------------
+
+For production deployments, it is recommended that all services use the built-in ``telepath`` SSL/TLS
+protocol.  You may deploy a service using TLS encryption by specifying a ``--telepath`` listen URL option, such
+as ``ssl://cortex.vertex.link/``.
+
+Under some circumstances, such as inbound DNAT networks or multi-homed hosts, it maybe necessary to specify a
```
Comment (jnwatson) [inline docs/synapse/devops.rst:19]: "maybe" -> may be

**Pull request merged**